docs/sources/operators-guide/running-production-environment/production-tips/index.md

```diff
@@ -84,6 +91,13 @@ The total number of file descriptors used to load index-headers linearly increas
 
 We recommend configuring the system's `file-max` ulimit at least to `65536` to avoid reaching the maximum number of open file descriptors.
 
+### Store-gateway disk IOPS
+
+The store-gateway disk performances (IOPS and latency) can impact queries.
+The store-gateway downloads block's [index-headers]({{< relref "../../architecture/binary-index-header.md" >}}) on local disk and reads them on each query hitting the long-term storage.
```

Comment:
```suggestion
The store-gateway downloads the block’s [index-headers]({{< relref "../../architecture/binary-index-header.md" >}}) onto local disk, and reads them on each query that reaches the long-term storage.
```
For any form of "to hit", we replace it with whatever it mean because "to hit" means "to strike".

What do you mean by "each query hitting the long-term storage"?
* "each query that reaches..."?
* "each query that accesses..."?
* "each query that calls from..."?
* "each query that needs to access..."?

Reply:
I mean "on each query that needs to fetch data from the long-term storage".

Reply:
We need another preposition besides "on" in the context of "on each query". What does "on" mean in this context?